src/Task/Runner/TaskRunner.php

```diff
@@ -114,6 +115,11 @@ public function run(TaskHandlerInterface $handler, TaskExecutionInterface $execu
 
         try {
             $execution = $this->hasPassed($execution, $this->handle($handler, $execution));
+        } catch (RetryException $exception) {
+            $execution = $this->taskExecutionRepository->findByUuid($execution->getUuid());
+            $this->taskExecutionRepository->save(clone $execution);
```

Comment:
will be called in the same run - which does not make much sense. should be scheduled but not executed (maybe call the scheduler)